falcon/request.py

```diff
@@ -1028,6 +1059,22 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_rfc_forwarded(self):
+        """rfc 7239"""
+        addr = []
+        for forwarded in self.env['HTTP_FORWARDED'].split(','):
+            for param in forwarded.split(';'):
+                param = param.strip().split('=', 1)
+                if len(param) == 1:
+                    continue
+                key, val = param
+                if key.lower() not in ('for', 'by'):
+                    # we only want for/by params
+                    continue
+                host, _ = parse_host(unquote_string(val))
```

Comment:
Indeed, it's not expressly forbidden by the RFC. It's interesting that werkzeug cheats a little bit here and doesn't actually try to unescape anything besides `"` and `\` characters. FWIW, the werkzeug code is about twice as fast as the list comprehension approach on my MBP.

Anyway, I guess we have three options:

(a) Don't try to unescape anything, only strip enclosing quotes. Simplest and most performant option. Some amount of risk, but likely minimal.

(b) Follow werkzeug's lead and only worry about the cases in which the RFC recommends senders SHOULD use escaping (for `"` and `\` chars). Fairly performant. But wouldn't actually be less risky than (a) in the case of the Forwarded header, since those two chars aren't used in addresses or obfuscated identifiers anyway.

(c) Strictly follow the RFC, and perhaps experiment with the implementation to see if we can make it a little bit more efficient.

Presumably, werkzeug has had pretty good luck with their strategy. It's not perfect, but is reasonably efficient. I'm guessing that senders tend not to escape characters unnecessarily, and that will carry over to their use of the new Forwarded header. Therefore, I would personally lean toward option (a). This is just my $0.02; happy to consider other opinions.